src/Model/Route.php

```diff
@@ -233,13 +237,11 @@ public function getPath()
      */
     public function setPath($pattern)
     {
-        $len = strlen($this->getStaticPrefix());
-
-        if (strncmp($this->getStaticPrefix(), $pattern, $len)) {
+        if (0 !== strpos($pattern, $this->getStaticPrefix())) {
             throw new \InvalidArgumentException('You can not set a pattern for the route that does not start with its current static prefix. First update the static prefix or directly use setVariablePattern.');
         }
 
-        return $this->setVariablePattern(substr($pattern, $len));
+        return $this->setVariablePattern(substr($pattern, $this->getStaticPrefix()));
```

Comment:
`$this->getStaticPrefix()` returns string, `substr` expects int - I guess it should be `strlen($this->getStaticPrefix())`.